### ISSUE TITLE
Autoclick when right-clicking on the editor

### ISSUE THREAD

**Issue by bogdanbeniaminb** [microsoft/vscode#115984]
Issue Type: <b>Bug</b>

I have a few extensions installed that create items in the context menu that appears when right-clicking on the editor content.

When I right-click on the editor somewhere in the middle of the page, so that the context menu cannot fit totally above or totally beneath the mouse cursor, the context menu is displayed in the middle of the screen, with the mouse cursor being somewhere at the middle of the context menu. The menu is slightly to the left (some 2px or so), so that the cursor is actually above the context menu. When I right-click (fast right-click, as you would normally do to open the context menu), the menu item that gets underneath the menu cursor automatically gets clicked. As a result, I have to right-click, move the mouse a bit to the left, and only afterwards release the mouse button. This makes the context menu very difficult to use. (EDIT: the cursor appeared above the menu items in previous versions of VSCode as well, but the menu items didn't get autoclicked. The menu would simply appear, sometimes with that menu item in a :hover state, but it didn't autoclick).

VS Code version: Code 1.53.0 (8490d3dde47c57ba65ec40dd192d014fd2113496, 2021-02-03T15:56:04.185Z)
OS version: Linux x64 5.4.89-1-MANJARO

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 2409)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|31.13GB (4.38GB free)|
|Process Argv|--unity-launch --enable-proposed-api liximomo.sftp --crash-reporter-id edf1c02d-a9f7-4b06-8353-259fa09af56a|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|/usr/share/xsessions/plasma|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (74)</summary>

Extension|Author (truncated)|Version
---|---|---
laravel-model-snippets|ahi|2.0.0
laravel-extra-intellisense|ami|0.4.1
vscode-browser-preview|auc|0.6.7
tailwind-docs|aus|1.4.0
lit-html|bie|1.11.1
vscode-intelephense-client|bme|1.6.3
vscode-tailwindcss|bra|0.5.8
better-phpunit|cal|1.5.3
npm-intellisense|chr|1.3.1
path-intellisense|chr|2.3.0
livewire-vscode|cie|0.2.8
gitignore|cod|0.6.0
dart-code|Dar|3.19.2
flutter|Dar|3.19.0
vscode-markdownlint|Dav|0.38.0
mustache|daw|1.1.1
vscode-eslint|dba|2.1.14
react-redux-react-router-snippets|dis|0.4.29
xml|Dot|2.5.1
es7-react-js-snippets|dsz|3.1.0
vscode-babel-coloring|dza|0.0.4
gitlens|eam|11.2.1
toggle-excluded-files|eam|1.7.0
EditorConfig|Edi|0.16.4
vscode-npm-script|eg2|0.3.13
apacheconf-snippets|eim|1.3.0
prettier-vscode|esb|5.9.1
git-project-manager|fel|1.7.1
php-pack|fel|1.0.2
terminal|for|0.0.10
todo-tree|Gru|0.0.196
headwind|hey|1.7.0
beautify|Hoo|1.5.0
smarty|imp|0.3.0
search-node-modules|jas|1.3.0
docthis|joe|0.7.1
vscode-wordpress-hooks|joh|0.5.3
vscode-styled-components|jpo|1.5.0
sftp|lix|10.12.10
vscode-language-babel|mgm|0.0.30
git-graph|mhu|1.28.0
dotenv|mik|1.0.1
theme-monokai-pro-vscode|mon|1.1.18
vscode-duplicate|mrm|1.2.1
vscode-scss|mrm|0.9.0
remote-ssh|ms-|0.63.0
remote-ssh-edit|ms-|0.63.0
sqltools|mtx|0.23.0
sqltools-driver-mysql|mtx|0.2.0
vscode-project-name-in-statusbar|muj|1.0.0
php-docblocker|nei|2.1.0
vetur|oct|0.32.0
vscode-php-getters-setters|php|1.2.3
material-icon-theme|PKi|4.5.0
vscode-react-refactor|pla|1.1.1
vscode-xml|red|0.15.0
vscode-yaml|red|0.14.0
create-file-folder|rit|0.0.5
relative-import|sai|1.3.1
vscode-javascript-booster|sbu|14.0.1
code-settings-sync|Sha|3.4.3
vscode-fileutils|sle|3.4.4
php-formatter|Sop|0.2.4
vscode-status-bar-format-toggle|tom|2.0.0
pdf|tom|1.1.0
wordpress-snippet|tun|1.1.5
vscode-todo-highlight|way|1.0.4
literally-html|web|0.1.2
html-snippets|wge|0.1.5
twig|wha|1.0.2
vscode-import-cost|wix|2.15.0
vscode-surround|yat|1.0.2
vscode-intelli-refactor|ypr|0.1.2
material-theme|zhu|3.9.13

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspyt653:30253241
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30211401
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt600cf:30251589

```

</details>

**Comment by bogdanbeniaminb** [comment]
![image](https://user-images.githubusercontent.com/45259752/107125659-e8da7e00-68b3-11eb-84ff-2e439b24cb16.png)

This is how the menu appears after right-clicking (I was keeping the mouse button pressed, so that it didn't autoclick). As you can see, the menu appears 1px or 2px to the left of the mouse cursor and this causes the autoclick.

**Comment by bogdanbeniaminb** [comment]
The same happens in the **insiders** version:


Issue Type: <b>Bug</b>

VS Code version: Code - Insiders 1.54.0-insider (52838cf6799cc448e738677ec37e86cf62a5bd89, 2021-02-05T05:16:37.087Z)
OS version: Linux x64 5.4.89-1-MANJARO

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 2365)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 3|
|Memory (System)|31.13GB (3.03GB free)|
|Process Argv|--no-sandbox --unity-launch|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|/usr/share/xsessions/plasma|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
GetterAndSetterGenerator|afm|1.0.3
vscode-intelephense-client|bme|1.6.3
prettier-vscode|esb|5.9.1
RelativePath|jak|1.4.0
vscode-duplicate|mrm|1.2.1
vscode-php-getter-setter|Myk|1.0.0
relative-import|sai|1.3.1
vscode-fileutils|sle|3.4.4


</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspyt653:30253241
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30208929
vstry914:30244316
pythonvsdeb440:30224570
pythonvsded773:30223139
pythonvspyt600cf:30251589
core-portspanel:30233467
coreuntitledfile:30249963

```

</details>

**Comment by bogdanbeniaminb** [comment]
Also, this is NOT happening with VS Code 1.52. In 1.52, even though the menu still appears to come under the mouse cursor, the item beneath it does NOT get autoclicked when doing a normal right-click. If I keep the mouse button pressed for a longer time, it does get autoclicked when I release the mouse button. But not in a normal (fast) right-click.

**Comment by yume-chan** [comment]
Duplicate of #113175, set `"window.titleBarStyle": "custom"` may help although it also changes your title bar.

**Comment by bogdanbeniaminb** [comment]
Yes, using custom menus/custom title bar does help, but it creates other issues for my workflow. One is that I can no longer hide the title bar, as I currently do with the native one (I'm using Linux/KDE). The other is that the menus are much, much larger, to a point where using them means scrolling through the list of options and this makes the workflow more difficult.

I understand some have had this issue years ago as well (I found a github issue opened 4 years ago), but for me personally it only appeared in VSCode 1.53. I reverted back to 1.52 and it's working fine again. It seems to be something related either to the Electron version used in 1.53 or to VSCode itself.

For me personally, using the "custom menus" workaround doesn't seem a solution. I would rather prefer this issue is investigated and fixed, as you would normally do with bugs.

Thank you.
